include/tiny-cuda-nn/encodings/grid.h

```diff
@@ -782,6 +1101,133 @@ class GridEncodingTemplated : public GridEncoding<T> {
+			return;
+		}
+
+		const auto& forward = dynamic_cast<const ForwardContext&>(ctx);
+
+		const T* dL_dy_rm = dL_doutput.data();
+
+		GPUMemoryArena::Allocation workspace;
+		if (dL_doutput.layout() == CM) {
+			workspace = allocate_workspace(stream, num_elements * m_n_features * sizeof(T));
+
+			// Transpose dL_dy. Use the buffer previously occupied by the encoded positions
+			const dim3 threads_transpose = { m_n_levels * N_FEATURES_PER_LEVEL, 8, 1 };
+			const uint32_t blocks_transpose = div_round_up(num_elements, threads_transpose.y);
+			transpose_gradients<T><<<blocks_transpose, threads_transpose, 0, stream>>>(
+				num_elements,
+				(T*)workspace.data(),
+				dL_doutput.pitched_ptr()
+			);
+
+			dL_dy_rm = (const T*)workspace.data();
+		}
+
+		if (param_gradients_mode != EGradientMode::Ignore) {
+			// We accumulate gradients with grad_t precision, which, for performance reasons, is not always T.
+			// If not, accumulate in a temporary buffer and cast later.
+			grad_t* grid_gradient;
+			GPUMemoryArena::Allocation grid_gradient_tmp;
+
+			if (!std::is_same<grad_t, T>::value) {
+				grid_gradient_tmp = allocate_workspace(stream, m_n_params * sizeof(grad_t));
+				grid_gradient = (grad_t*)grid_gradient_tmp.data();
+			} else {
+				grid_gradient = (grad_t*)m_grid_gradient;
+			}
+
+			if (param_gradients_mode == EGradientMode::Overwrite) {
+				CUDA_CHECK_THROW(cudaMemsetAsync(grid_gradient, 0, n_params() * sizeof(grad_t), stream));
+			}
+
+			static constexpr uint32_t N_THREADS_HASHGRID = 256;
+			static constexpr uint32_t N_FEATURES_PER_THREAD = std::min(2u, N_FEATURES_PER_LEVEL);
+
+			const dim3 blocks_hashgrid = { div_round_up(num_elements * N_FEATURES_PER_LEVEL / N_FEATURES_PER_THREAD, N_THREADS_HASHGRID), m_n_levels, 1 };
+
+			// from dL_d(dL_dx) to dL_dgrid
+			kernel_grid_backward_input_backward_grid<T, grad_t, N_POS_DIMS, N_FEATURES_PER_LEVEL, N_FEATURES_PER_THREAD><<<blocks_hashgrid, N_THREADS_HASHGRID, 0, stream>>>(
+				num_elements,
+				m_n_features,
+				m_hashmap_offsets_table.data(),
+				m_base_resolution,
+				std::log2(m_per_level_scale),
+				this->m_max_level,
+				this->m_max_level_gpu,
+				m_interpolation_type,
+				m_grid_type,
+				// inputs
+				dL_ddLdinput.view(),
+				forward.positions.data() ? forward.positions.view() : input.view(), // positions SoA
+				dL_dy_rm, // gradients SoA
+				// outputs
+				grid_gradient
+			);
+
+			if (!std::is_same<grad_t, T>::value) {
+				parallel_for_gpu(stream, n_params(), [grad=m_grid_gradient, grad_tmp=grid_gradient] __device__ (size_t i) {
+					grad[i] = (T)grad_tmp[i];
+				});
+			}
+		}
+
+		if (dL_ddLdoutput) {
+			// from dL_d(dL_dx) to dL_doutput
+			linear_kernel(kernel_grid_backward_input_backward_dLdoutput<T, N_POS_DIMS>, 0, stream, 
+				num_elements,
+				m_n_features,
+				// inputs
+				dL_ddLdinput.view(),
+				forward.dy_dx.data(),
+				dL_dy_rm,
+				// outputs
+				dL_ddLdoutput->pitched_ptr()
```

Comment:
Could you replace this pitched pointer with `->view()` in order to also support row-major `dL_ddLdoutput`?